File: MultiLanguageSandbox/src/test/deps/cpp/27.cpp
#include <cctype>
#include <cassert>
#include <cstdio>
/**
 * Transforms the case of a given letter.
 * 
 * If the input is a lowercase letter, it returns the uppercase version,
 * and if the input is an uppercase letter, it returns the lowercase version.
 * Non-alphabetic characters are returned unchanged.
 *
 * Usage:
 *   char result = transform_letter_case('b'); // result will be 'B'
 *   char result = transform_letter_case('B'); // result will be 'b'
 *
 * @param letter The input letter to be transformed.
 * @return The letter with its case reversed if it's alphabetic, otherwise the original letter.
 */
// for islower, isupper, tolower, toupper

char transform_letter_case(char letter) {
    if (islower(letter)) {
        return toupper(letter);
    } else if (isupper(letter)) {
        return tolower(letter);
    }
    return letter;
}
int main() {
    assert(transform_letter_case('b') == 'B');
    assert(transform_letter_case('B') == 'b');
    // Additional test cases
    assert(transform_letter_case('z') == 'Z');
    assert(transform_letter_case('Z') == 'z');
    assert(transform_letter_case('m') == 'M');
    assert(transform_letter_case('M') == 'm');
    assert(transform_letter_case('1') == '1'); // Non-alphabetic, should be unchanged
    assert(transform_letter_case('!') == '!'); // Non-alphabetic, should be unchanged

    // Uncomment the following line to print a success message (optional)
    
    return 0;
}

File: MultiLanguageSandbox/src/test/deps/javascript/27.js

/**
 * Transforms the case of a given letter.
 * - letter (char): The input letter to be transformed.
 * 
 * Returns:
 * - char: The letter with its case reversed. If the input is lowercase,
 * returns uppercase, and if the input is uppercase, returns lowercase.
 * 
 * Example:
 * transformLetterCase('b') returns 'B'
 * transformLetterCase('Z') returns 'z'
 */

function transformLetterCase(letter) {
    if (letter === letter.toLowerCase() && letter !== letter.toUpperCase()) {
        return letter.toUpperCase();
    } else if (letter === letter.toUpperCase() && letter !== letter.toLowerCase()) {
        return letter.toLowerCase();
    }
    return letter; // Return as-is if it's not a letter (though the problem implies it is)
}
const testTransformLetterCase = () => {
    console.assert(transformLetterCase('b') === 'B');
    console.assert(transformLetterCase('B') === 'b');
    // Additional test cases
    console.assert(transformLetterCase('z') === 'Z');
    console.assert(transformLetterCase('Z') === 'z');
    console.assert(transformLetterCase('m') === 'M');
    console.assert(transformLetterCase('M') === 'm');
    console.assert(transformLetterCase('1') === '1'); // Non-alphabetic, should be unchanged
    console.assert(transformLetterCase('!') === '!'); // Non-alphabetic, should be unchanged

    // Uncomment the following line to log a message for passed tests
    // console.log("All tests passed successfully.");
};

testTransformLetterCase();